File: java/src/civitas/Sorpresa.java
package civitas;
import java.util.ArrayList;

public class Sorpresa {
    protected String texto;

    Sorpresa(String texto) {
        this.texto = texto;
    }

    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos) {
        return (actual >= 0 && actual < todos.size());
    }

    private void informe(int actual, ArrayList<Jugador> todos) {
        Diario.getInstance().ocurreEvento("Se está aplicando una sorpresa a: " + todos.get(actual).getNombre());
    }
}

File: ruby/lib/sorpresa.rb
require_relative("mazo_sorpresas")

module Civitas
    class Sorpresa

        def initialize(texto)
            @texto=texto
        end

        def aplicarAJugador(actual, todos)
        end

       
        def informe(actual, todos)
            Diario.instance.ocurreEvento("Se aplica #{toString} a "+todos[actual].nombre)
        end

        def jugadorCorrecto (actual, todos)
            return (actual >= 0 && actual <= todos.length)
        end

        def toString
            return "Sorpresa: #{@texto}"
        end

        private_class_method :new
    end
end